.github/dependabot.yml

```diff
@@ -5,19 +5,62 @@ registries:
     url: https://plugins.gradle.org/m2
     username: dummy # Required by dependabot
     password: dummy # Required by dependabot
+  maven-central:
+    type: maven-repository
+    url: https://repo1.maven.org/maven2/
+    username: dummy # Required by dependabot
+    password: dummy # Required by dependabot
 updates:
-  - package-ecosystem: "gradle"
+  - package-ecosystem: "github-actions"
     directory: "/"
     target-branch: "1.0.x" # oldest OSS supported branch
-    allow:
-      - dependency-name: "com.gradle*"
-    registries:
-      - gradle-plugin-portal
     schedule:
       interval: "weekly"
-    open-pull-requests-limit: 10
   - package-ecosystem: "github-actions"
     directory: "/"
-    target-branch: "1.0.x" # oldest OSS supported branch
+    target-branch: "1.1.x" # oldest OSS supported branch
+    schedule:
+      interval: "weekly"
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    target-branch: "main"
     schedule:
       interval: "weekly"
+  - package-ecosystem: gradle
+    directory: /
+    schedule:
+      interval: daily
+    target-branch: 1.0.x
+    ignore:
+      # only upgrade patch versions for maintenance branch
+      - dependency-name: "*"
+        update-types:
+          - version-update:semver-major
+          - version-update:semver-minor
+    registries:
+      - gradle-plugin-portal
+      - maven-central
```

Comment:
Could you add a "build dependencies" config like was done in docs-generator so we upgrade patch and minor versions of build dependencies? See https://github.com/micrometer-metrics/micrometer-docs-generator/blob/3ac3c2fd66a88acd3527b1627eb3d905aa6cb4d3/.github/dependabot.yml#L19-L49

You'll need to update the non-build dependencies config to be for directory `/gradle`. And you don't need to declare any registries for them because non-build dependencies should all come from maven-central, which is the default.